app/src/Vues/Register.jsx

```diff
@@ -0,0 +1,201 @@
+import React, {useState} from "react";
+import Input from "../components/Utils/Input.jsx";
+import {Link} from "react-router-dom";
+import {AlertError, AlertSuccess} from "../components/Utils/Alert.jsx";
+import PasswordInput from "../components/Account/PasswordInput.jsx";
+import UserModel from "../scripts/Models/UserModel.ts";
+import ErrorResponse from "../scripts/API/Responses/ErrorResponse.ts";
+import '../index.css';
+
+function Register () {
+
+    const [email, setEmail] = useState('');
+    const [prenom, setPrenom] = useState('');
+    const [nom, setNom] = useState('');
+    const [password, setPassword] = useState('');
+    const [confirmPassword, setConfirmPassword] = useState('');
+    const [login, setLogin] = useState('');
+
+    const [emailError, setEmailError] = useState('');
+    const [prenomError, setprenomError] = useState('');
+    const [nomError, setnomError] = useState('');
+    const [passwordError, setPasswordError] = useState('');
+    const [confirmPasswordError, setConfirmPasswordError] = useState('');
+    const [loginError, setLoginError] = useState('');
+
+    const [generalError, setGeneralError] = useState("");
+    const [success, setSuccess] = useState(false);
+
+    const handleSignUp = async () => {
+                const userData = {
+                    id : 0,
+                    login: login,
+                    firstname: prenom,
+                    lastname: nom,
+                    mail: email,
+                };
+
+                setSuccess(true)
+
+                try {
+
+                    const userModel = new UserModel(userData);
+                    userModel.password = password
+                    userModel.password_confirm = confirmPassword
+
+                    const response = await userModel.createUser();
+
+                    if (response instanceof ErrorResponse) {
+                        setSuccess(false)
+                        if (response.errorCode() === 401) {
+                            setGeneralError("Identifiants incorrects");
+                        } else {
+                            setGeneralError(`Une erreur est survenue: ${response.errorMessage()}`);
+                        }
+                        return;
+                    }
+
+                } catch (err) {
+                    setSuccess(false)
+                    const errorMessage = err instanceof Error ? err.message : "Une erreur est survenue";
+                    setGeneralError(`Une erreur est survenue: ${errorMessage}`);
+                }
+    };
+
+    const handleMailType = (e) => {
```

Comment:
You should probably place this method into a utils script. Might be useful for other components.

Reply:
en vrai, hormis dans le Form on va pas les utiliser beaucoup donc je pense les laisser dans le form en vrai